src/ir/visitor.cpp

```diff
@@ -95,13 +95,20 @@ bool dfs_ir_post_order_visitor::visit_strategy(node &node)
 
 bool bfs_ir_pre_order_visitor::visit_strategy(node &node)
 {
-    std::queue<nncase::ir::node *> nodes;
-    nodes.push(&node);
+    std::deque<nncase::ir::node *> nodes;
+    nodes.push_back(&node);
+
+    auto added = [&](nncase::ir::node *node) {
```

Comment:
Use an `unordered_set` to check whether the node has been queued.

Reply:
modify as advised.